src/components/ProfilePopover/index.tsx

```diff
@@ -1,78 +1,78 @@
 import { useEffect, useState } from "react";
+import Image from "next/image";
 import { UserResource } from "@clerk/types";
 import { Popover, Transition } from "@headlessui/react";
 import { GoCheck, GoPencil, GoX } from "react-icons/go";
 
 import { useAuth } from "@/hooks/auth";
-import { UserRole } from "@/types";
 
-interface Props {
-  firstName: string | null | undefined;
-  lastName: string | null | undefined;
-  role: UserRole;
-  desc: string | null | undefined;
+async function updateUserMetadata(user: UserResource, description: string) {
+  await user.update({ unsafeMetadata: { description } });
 }
 
-const updateUserMetadata = async (
-  user: UserResource | null | undefined,
-  description: string
-) => {
-  await user?.update({
-    unsafeMetadata: {
-      description: description
-    }
-  });
-};
+export default function ProfilePopover() {
+  const { user, role } = useAuth();
 
-export default function ProfilePopover({
-  firstName,
-  lastName,
-  role,
-  desc
-}: Props) {
   const [isEdit, setIsEdit] = useState(false);
-  const [description, setDescription] = useState(desc);
-  const [initialDescription, setInitialDescription] = useState(desc);
-  const { user } = useAuth();
-
-  // if (firstName === null || firstName === undefined) return null;
+  const [description, setDescription] = useState(
+    String(user?.unsafeMetadata.description) || ""
+  );
+  const [initialDescription, setInitialDescription] = useState(description);
 
   useEffect(() => {
-    setDescription(desc);
-  }, [desc]);
+    setDescription(String(user?.unsafeMetadata.description) || "");
+  }, [user?.unsafeMetadata.description]);
 
   const handleChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
     const newValue = e.target.value;
     if (newValue.length <= 200) {
-      setDescription(e.target.value); // Update the description state with the new value
+      setDescription(newValue);
     }
   };
```

Comment:
you can use zod to validate the input

Reply:
or use React Hook Form

Reply:
Good points, wasn't really thinking about refactoring the validation.

Maybe for another issue?